prow/git/git.go

```diff
@@ -323,17 +312,23 @@ func (r *Repo) CheckoutPullRequest(number int) error {
 // Config runs git config.
 func (r *Repo) Config(key, value string) error {
 	r.logger.Infof("Running git config %s %s", key, value)
-	if b, err := r.gitCommand("config", key, value).CombinedOutput(); err != nil {
+	if b, err := r.gitCommand("config", key, value); err != nil {
 		return fmt.Errorf("git config %s %s failed: %v. output: %s", key, value, err, string(b))
 	}
 	return nil
 }
 
 // Remote builds a remote url from user, pasword, and a slice of path items
-func Remote(base *url.URL, user string, pass string, pathItems ...string) (*url.URL, error) {
-	base.User = url.UserPassword(user, pass)
-	base.Path = strings.Join(pathItems, "/")
-	return base, nil
+func Remote(base *url.URL, user string, pass string, pathItems ...string) *url.URL {
+	copy := *base
+	if user != "" && pass != "" {
+		copy.User = url.UserPassword(user, pass)
+	}
+	if len(pathItems) != 0 {
+		fullpath := append([]string{copy.Path}, pathItems...)
```

Comment:
This is the error, right? Aka previously we did `base.Path = strings.Join(pathItems, "/")` and now we're essentially doing `base.Path += strings.Join(pathItems, "/")`

Reply:
This is not. This is better equivalent of what it used to be:
https://github.com/kubernetes/test-infra/blob/e126eb05def4c0091c97623895ba6726b51fa98b/prow/git/git.go#L150
https://github.com/kubernetes/test-infra/blob/e126eb05def4c0091c97623895ba6726b51fa98b/prow/git/git.go#L300